### PR TITLE
[Accelerator 4] Adding event notification endpoint implementation

### PR DESCRIPTION
**Pull request by Ashi1993**
## Adding Event Notification Endpoint Implementation

> This PR add Event Notification Endpoint Implementation

**Issue link:** *https://github.com/wso2-enterprise/solutions-bfsi/issues/20*

**Doc Issue:** *Optional, link issue from [documentation repository](https://github.com/wso2/docs-ob/issues)*

**Applicable Labels:** *Spec, product, version, type (specify requested labels)*

------

### Development Checklist

1. [x] Build complete solution with pull request in place.
2. [x] Ran checkstyle plugin with pull request in place.
3. [x] Ran Findbugs plugin with pull request in place.
4. [x] Ran FindSecurityBugs plugin and verified report.
5. [x] Formatted code according to WSO2 code style.
6. [x] Have you verified the PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?
7. [ ] Migration scripts written (if applicable).
8. [x] Have you followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)?

### Testing Checklist

1. [ ] Written unit tests.
2. [ ] Verified tests in multiple database environments (if applicable).
3. [ ] Tested with BI enabled  (if applicable).

### CODE REVIEW

File: financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/util/EventSubscriptionUtils.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+import java.io.IOException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Response;
+
+/**
+ * Events Notification Subscription API Utils.
+ */
+public class EventSubscriptionUtils {
```
Comment (VimukthiRajapaksha) [inline financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/util/EventSubscriptionUtils.java:37]: Let's add a private constructor. applicable to other relevant places too

Comment (Ashi1993) [reply financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/util/EventSubscriptionUtils.java:37]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/169/commits/1f79f1298f086fcd945a3516570d04781c3fec0e

File: financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventPollingEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+/**
+ * Aggregated Event Polling API Specification.
+ *
+ * <p>Swagger for Aggregated Event Polling API Specification
+ */
+@Path("/events")
+public class EventPollingEndpoint {
+
+    private static final Log log = LogFactory.getLog(EventCreationEndpoint.class);
+    private static final EventPollingServiceHandler eventPollingServiceHandler = EventNotificationUtils.
```
[the remark this answers is not in the document]
Comment (Ashi1993) [reply financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventPollingEndpoint.java]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/169/commits/1f79f1298f086fcd945a3516570d04781c3fec0e

File: financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventPollingEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+/**
+ * Aggregated Event Polling API Specification.
+ *
+ * <p>Swagger for Aggregated Event Polling API Specification
+ */
+@Path("/events")
+public class EventPollingEndpoint {
+
+    private static final Log log = LogFactory.getLog(EventCreationEndpoint.class);
```
[the remark this answers is not in the document]
Comment (Ashi1993) [reply financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventPollingEndpoint.java]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/169/commits/1f79f1298f086fcd945a3516570d04781c3fec0e

File: financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventCreationEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+            return EventNotificationUtils.mapEventCreationServiceResponse(eventCreationResponse);
+
+        } catch (ClassCastException e) {
+            log.error(EventNotificationEndPointConstants.REQUEST_PAYLOAD_ERROR, e);
+            return Response.status(Response.Status.BAD_REQUEST).entity(EventNotificationServiceUtil
+                    .getErrorDTO(EventNotificationEndPointConstants.INVALID_REQUEST_PAYLOAD,
+                            EventNotificationEndPointConstants.REQUEST_PAYLOAD_ERROR)).build();
+        } catch (FSEventNotificationException e) {
+            return Response.status(e.getStatus()).entity(EventNotificationServiceUtil
```
[the remark this answers is not in the document]
Comment (Ashi1993) [reply financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventCreationEndpoint.java:155]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/169/commits/1f79f1298f086fcd945a3516570d04781c3fec0e

File: financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventCreationEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                notificationCreationDTO.setResourceId(request.getHeader(
+                        EventNotificationEndPointConstants.X_WSO2_RESOURCE_ID));;
+            } else {
+                return Response.status(Response.Status.BAD_REQUEST).entity(EventNotificationServiceUtil.
+                        getErrorDTO(EventNotificationEndPointConstants.MISSING_REQUEST_HEADER,
+                        EventNotificationConstants.MISSING_HEADER_PARAM_RESOURCE_ID)).build();
+            }
+
+            //set events to notificationCreationDTO
+            JSONObject finalNotificationEvents = notificationEvents;
```
[the remark this answers is not in the document]
Comment (Ashi1993) [reply financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventCreationEndpoint.java]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/169/commits/1f79f1298f086fcd945a3516570d04781c3fec0e

File: financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventCreationEndpoint.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+            notificationEvents.keySet().forEach(eventName -> {
+                JSONObject eventInformation = (JSONObject) finalNotificationEvents.get(eventName);
```
[the remark this answers is not in the document]
Comment (Ashi1993) [reply financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventCreationEndpoint.java]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/169/commits/1f79f1298f086fcd945a3516570d04781c3fec0e

File: financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventCreationEndpoint.java [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+        NotificationCreationDTO notificationCreationDTO = new NotificationCreationDTO();
+        String requestData = StringUtils.EMPTY;
+        JSONObject notificationEvents;
+
+        try {
+            //Check if the request pay load is empty
+            if (!parameterMap.isEmpty() && parameterMap.containsKey(EventNotificationEndPointConstants.REQUEST)) {
+
+                requestData = parameterMap.get(EventNotificationEndPointConstants.REQUEST).
+                        toString().replaceAll("\\\\r|\\\\n|\\r|\\n|\\[|]| ", StringUtils.EMPTY);
```
[the remark this answers is not in the document]
Comment (Ashi1993) [reply financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.event.notifications.endpoint/src/main/java/org/wso2/financial/services/accelerator/event/notifications/endpoint/api/EventCreationEndpoint.java]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/169/commits/1f79f1298f086fcd945a3516570d04781c3fec0e

**Review by anjuchamantha — APPROVED** [review]

**Review by VimukthiRajapaksha — APPROVED** [review]

**Pull request merged**